kube/deployment.go

```diff
@@ -19,20 +20,48 @@ func NewDeployment(role *model.Role, settings *ExportSettings) (helm.Node, helm.
 		return nil, nil, err
 	}
 	spec := helm.NewEmptyMapping()
-	spec.AddNode("selector", newSelector(role.Name))
-	spec.AddNode("template", podTemplate)
+	spec.Add("selector", newSelector(role.Name))
+	spec.Add("template", podTemplate)
 
 	deployment := newKubeConfig("extensions/v1beta1", "Deployment", role.Name, helm.Comment(role.GetLongDescription()))
-	replicaCheck(role, deployment, spec, svc, settings)
-	deployment.AddNode("spec", spec.Sort())
+	err = replicaCheck(role, deployment, spec, svc, settings)
+	deployment.Add("spec", spec.Sort())
```

Comment:
Maybe `spec` should be added to `deployment` first and not passed as an explicit parameter to `replicaCheck`. The function already locates other fields inside `spec` via `Get()` calls, so why not locate `spec` itself too and get rid of the redundant parameter?

Reply:
https://github.com/SUSE/fissile/pull/282/commits/27d37c3de22062e27a141fd59731071c3561ee30